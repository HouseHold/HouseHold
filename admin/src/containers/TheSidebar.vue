<template>
  <CSidebar 
    fixed 
    :minimize="minimize"
    :show.sync="show"
  >
    <CSidebarBrand 
      :imgFull="{ width: 118, height: 46, alt: 'Logo', src: 'img/brand/coreui-base-white.svg'}"
      :imgMinimized="{ width: 118, height: 46, alt: 'Logo', src: 'img/brand/coreui-signet-white.svg'}"
      :wrappedInLink="{ href: 'https://coreui.io/', target: '_blank'}"
    />
    <CRenderFunction flat :content-to-render="nav"/>
    <CSidebarMinimizer
      class="d-md-down-none"
      @click.native="minimize = !minimize"
    />
  </CSidebar>
</template>

<script>
import nav from './_nav'

export default {
  name: 'TheSidebar',
  data () {
    return {
      minimize: false,
      nav,
      show: 'responsive'
    }
  },
  mounted () {
    this.$root.$on('toggle-sidebar', () => {
      const sidebarOpened = this.show === true || this.show === 'responsive'
      this.show = sidebarOpened ? false : 'responsive'
    })
    this.$root.$on('toggle-sidebar-mobile', () => {
      const sidebarClosed = this.show === 'responsive' || this.show === false
      this.show = sidebarClosed ? true : 'responsive'
    })
  }
}
</script>
