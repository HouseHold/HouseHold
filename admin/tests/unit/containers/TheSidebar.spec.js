import Vue from 'vue'
import { shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import TheSidebar from '@/containers/TheSidebar'

Vue.use(CoreuiVue)

describe('TheSidebar.vue', () => {
  it('has a name', () => {
    expect(TheSidebar.name).toBe('TheSidebar')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(TheSidebar)
    expect(wrapper.element).toMatchSnapshot()
  })
})
