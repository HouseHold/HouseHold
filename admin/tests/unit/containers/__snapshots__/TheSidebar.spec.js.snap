// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidebar.vue renders correctly 1`] = `
<csidebar-stub
  breakpoint="lg"
  colorscheme="dark"
  dropdownmode="openActive"
  fixed="true"
  hideonmobileclick="true"
  show="responsive"
>
  <csidebarbrand-stub
    imgfull="[object Object]"
    imgminimized="[object Object]"
    wrappedinlink="[object Object]"
  />
   
  <crenderfunction-stub
    contenttorender="[object Object]"
    flat="true"
  />
   
  <csidebarminimizer-stub
    class="d-md-down-none"
  />
</csidebar-stub>
`;
